format/FormatNumber.ts

```diff
@@ -28,73 +38,87 @@ const UP_TICK = '▴',
         neutral: 'xh-neutral-val'
     };
 
+export type Precision = 'auto' | IntRange<0, 13>;
+export type ZeroPad = boolean | IntRange<1, 13>;
```

Comment:
We can decide if we want the range - wanted to try it out but could be annoying for apps that have a dynamic value they are looking to pass that is typed as `number` would require cast